### DISCUSSION TITLE
How to navigate the route to not found using new app router

### DISCUSSION THREAD

**Discussion by karthik-js** [vercel/next.js#50838] [Help] [answered]
### Summary

We are converting our app from page router to app router incrementally. We depend heavily on server-side data fetching i.e. getServersideProps. The new app router has removed the data fetching using the functions in favor of fetch. 

When the API response from our server is used to respond with 404 or any specific error code. We used to leverage the notFound flag in getServersideProps return props.

I was looking for something similar with new data fetching if I have to show not found page manually. What would be the best approach with an app router?

**Comment by ciruz** [comment, marked as answer]
You can just fire `notFound()` to throw an error and render the `not-found` file, it even sets `<meta name="robots" content="noindex" />` for you.

https://nextjs.org/docs/app/api-reference/functions/not-found
https://nextjs.org/docs/app/api-reference/file-conventions/not-found
